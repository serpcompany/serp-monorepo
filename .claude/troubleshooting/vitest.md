https://vitest.dev/guide/common-errors.html