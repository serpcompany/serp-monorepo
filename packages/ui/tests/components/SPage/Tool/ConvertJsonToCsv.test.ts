import { mockNuxtImport } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'
import SPageToolConvertJsonToCsv from '../../../../components/SPage/Tool/ConvertJsonToCsv.vue'
import ComponentRender from '../../../componentRender'

mockNuxtImport('useSeoMeta', () => () => {})

describe('sPageToolConvertJsonToCsv Snapshot', () => {
  it('renders the default snapshot correctly', async () => {
    const html = await ComponentRender(
      'SPageToolConvertJsonToCsv',
      {},
      SPageToolConvertJsonToCsv,
    )
    expect(html).toMatchSnapshot()
  })
})
