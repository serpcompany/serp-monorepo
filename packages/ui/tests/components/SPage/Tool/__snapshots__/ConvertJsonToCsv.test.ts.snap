// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPageToolConvertJsonToCsv Snapshot > renders the default snapshot correctly 1`] = `
<div class="container">
  <div class="
    md:px-8
    md:py-20
    px-4
    py-10
    sm:px-6
    sm:py-16
  ">
    <section class="py-20">
      <h1 class="md:text-center">
        JSON to CSV Converter
      </h1>
      <p class="
        md:text-center
        sm:text-md
        text-base
        text-left
      ">
        Convert JSON to CSV format (comma separated list).
      </p>
    </section>
    <!-- in box -->
    <div class="
      flex
      gap-8
      justify-around
    ">
      <div class="
        inline-flex
        items-center
        mb-10
        relative
        w-full
      ">
        <textarea
          class="
            bg-default
            border-0
            disabled:cursor-not-allowed
            disabled:opacity-75
            focus-visible:ring-2
            focus-visible:ring-inset
            focus-visible:ring-primary
            focus:outline-none
            gap-1.5
            placeholder:text-dimmed
            px-2.5
            py-1.5
            ring
            ring-accented
            ring-inset
            rounded-md
            text-highlighted
            text-sm
            transition-colors
            w-full
          "
          placeholder="Paste your content here"
          rows="15"
          value=""
        ></textarea>
        <!-- v-if -->
        <!-- v-if -->
      </div>
      <!-- out box -->
      <div class="
        inline-flex
        items-center
        mb-10
        relative
        w-full
      ">
        <textarea
          class="
            bg-default
            border-0
            disabled:cursor-not-allowed
            disabled:opacity-75
            focus-visible:ring-2
            focus-visible:ring-inset
            focus-visible:ring-primary
            focus:outline-none
            gap-1.5
            placeholder:text-dimmed
            px-2.5
            py-1.5
            ring
            ring-accented
            ring-inset
            rounded-md
            text-highlighted
            text-sm
            transition-colors
            w-full
          "
          rows="15"
          value=""
        ></textarea>
        <!-- v-if -->
        <!-- v-if -->
      </div>
    </div>
    <!-- button -->
    <button
      class="
        aria-disabled:bg-primary
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-primary
        disabled:bg-primary
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:outline-2
        focus-visible:outline-offset-2
        focus-visible:outline-primary
        font-medium
        gap-1.5
        hover:bg-primary/75
        inline-flex
        items-center
        px-2.5
        py-1.5
        rounded-md
        text-inverted
        text-sm
        transition-colors
      "
      type="button"
    >
      <!-- v-if -->
      Submit
      <!-- v-if -->
    </button>
  </div>
</div>
`;
